### PR TITLE
Create article list - US1

### LINKED ISSUE #1: 1 - create single page with articles
## Value proposition

**As a user**
**I want to** see all offered articles
**so that** I can find a list with articles I want to buy.
## Description

Image

## Acceptance Criteria

- the site should  has a header with the app name
- all article cards should displayed in a list
- the latest article is on the top
- every card has a image and a few details of the articles (image, article name, size, gender, price)

## Tasks

- [x] create `Header` component
- [x] create a fake data file (db.json)
- [x] create `Article` component with image, article name, size, gender, price
- [x] fetch the fake data and display it on the page

## Complexity

M

### PR DESCRIPTION
**Pull request by EugenMoser**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
-  return <h1>Hello Flamingos! 🦩</h1>;
+import ArticleList from "../components/ArticleList";
+import styled from "styled-components";
+import useSWR from "swr";
+import { fetcher } from "../helpers/api";
+//import articlesDb from "../helpers/db.json";
+
+function Home() {
+  const { data: articles, error } = useSWR("/api/articles", fetcher);
+  console.log(articles);
```
Comment (Kheirah) [inline pages/index.js]: Consider removing console logs.

File: pages/index.js [hunk 2]
```diff
@@ -1,3 +1,28 @@
-export default function Home() {
-  return <h1>Hello Flamingos! 🦩</h1>;
+import ArticleList from "../components/ArticleList";
+import styled from "styled-components";
+import useSWR from "swr";
+import { fetcher } from "../helpers/api";
+//import articlesDb from "../helpers/db.json";
```
Comment (Kheirah) [inline pages/index.js]: Don't leave commented lines of code in your code😃

File: package.json [hunk 3]
```diff
@@ -17,7 +17,9 @@
     "next": "13.0.3",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "styled-components": "^5.3.6"
+    "styled-components": "^5.3.6",
+    "swr": "^1.3.0",
+    "use-router": "^0.4.2"
```
Comment (Kheirah) [inline package.json:22]: I'm wondering, what is this package `use-router` and what is it for? I don't think you need it. Did you install it by accident?

**Review by Kheirah — APPROVED** [review]
Very good!

Have a peek at my comments below.

File: components/Article/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  border: 1px solid black;
+  margin: 2rem;
+  list-style: none;
+`;
+
+const Art = styled.article`
+  display: flex;
+  width: 80vw;
+  height: 200px;
+`;
```
Comment (JessicaLoers) [inline components/Article/index.js]: I notice the following things you could do to make your code cleaner and your html structure more semantic. 

- `<List>` / `<li>`: This is a list element without a parent `<ul>` or `<ol>` 😲 - actually you don't need this parent element anyway. 
- ``: It's more than  recommended to use the [next/image component: documentation](https://nextjs.org/docs/api-reference/next/image). 
- I recommend to choose a more descriptive name for your `<Art>` component e.g. `<StyledArticle>`
- you could change the `<div>` into a `<ul>` with `<li>`
- Please have a look at my comments below 🫣

```suggestion
import styled from "styled-components";
// hint: import the next Image component (see Link above: next.js documentation)

function Article({ article }) {
 
 // just a suggestion for better readability. by the way: I recommend to change the key "article" into "name". 
 const { article, size, gender, price } = article

  return (
      <StyledArticle>
        <Image
        // with needed properties (see Link above: next.js documentation)
        />
        <h3>{article}</h3>
        <ul>
          <li>Größe: {size}</li>
          <li>Geschlecht: {gender}</li>
          <li>Preis: {price}</li>
        </ul>
      </StyledArticle>
  );
}

export default Article;

// Don't forget to adjust your styled components according to the changes.
const List = styled.li`
  border: 1px solid black;
  margin: 2rem;
  list-style: none;
`;

const Art = styled.article`
  display: flex;
  width: 80vw;
  height: 200px;
`;
```

File: components/ArticleList/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    <>
+      {articles.map((article) => {
+        return (
+          <Article
+            key={article.id}
+            article={article}
+          />
+        );
+      })}
+    </>
```
Comment (JessicaLoers) [inline components/ArticleList/index.js]:
```suggestion
  <>
  // you can render the 
    <h2>Alle Artikel in deiner Nähe:</h2>
    <StyledList>
        {articles.map(article => {
          return (
            <li key={article.id} >
              <Article
                article={article}
              />
            </li>
          );
        })}
      </StyledList>
    </>
    
const StyledList = styled.ul `
// ....
`   
```

File: helpers/db.json [hunk 6]
```diff
@@ -0,0 +1,66 @@
+[
+  {
+    "id": 1,
+    "article": "Body",
```
Comment (JessicaLoers) [inline helpers/db.json]: I recommend to rename this key 👇

```suggestion
    "name": "Body",
```

File: helpers/db.json [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    "price": 10,
```
Comment (JessicaLoers) [inline helpers/db.json]: I recommend to add decimal digits 👇
```suggestion
    "price": 10.00,
```

File: pages/index.js [hunk 8]
```diff
@@ -1,3 +1,28 @@
...
+    return <h4>loading...</h4>;
+  }
+  return (
+    <>
+      <h2>Alle Artikel in deiner Nähe:</h2>
+      <UL>
+        <ArticleList articles={articles} />
+      </UL>
+    </>
+  );
```
Comment (JessicaLoers) [inline pages/index.js]: I suggest to move the `<h2>` into the `<ArticleList>` (see comments in ArticleList components). 
Apart from the fact that `<UL>` is a very bad name, you no longer need this element. 
Now it's super clean 🧽

```suggestion
  return (
        <ArticleList articles={articles} />
  );
```

**Review by JessicaLoers — CHANGES_REQUESTED** [review]
Hi @EugenMoser ✌️

as mentioned in the Daily this morning I took a closer look at your code.  
Overall, super duper good work. 

Regarding the <Article> and <ArticleList> components. There are some structural suggestions for improvement that I would like to recommend to you and that hopefully will make your life easier in the long run. 

**Check out my comment and please let me know if you have any questions**

File: pages/index.js (hunk 8 above)
Comment (EugenMoser) [reply pages/index.js]: thanks for your very good feedback. I fixed it.

File: pages/index.js [hunk 9]
```diff
@@ -2,27 +2,20 @@ import ArticleList from "../components/ArticleList";
...
     return <h4>loading...</h4>;
   }
   return (
     <>
-      <h2>Alle Artikel in deiner Nähe:</h2>
-      <UL>
+      <main>
         <ArticleList articles={articles} />
-      </UL>
+      </main>
```
Comment (JessicaLoers) [inline pages/index.js]: Nice ...I forgot the missing `<main>` 🎉

**Review by JessicaLoers — APPROVED** [review]

**Comment by JessicaLoers** [comment]
@EugenMoser Well done. You'r good to go to the next step... the **Quality Assurance**  🥳

**Pull request merged**